### ISSUE TITLE
NilClass error in Compliance Phase when no client.pem present

### ISSUE THREAD

**Issue by tas50** [chef/chef#11219]
When running chef-client w/o a client.pem we get the usual missing client.pem file error as well as a NilClass error from the Compliance Phase we shouldn't be seeing.

```
Running handlers:
[2021-03-23T18:27:01-07:00] ERROR: Running exception handlers
Running handlers complete
[2021-03-23T18:27:01-07:00] ERROR: Exception handlers complete
Chef Infra Client failed. 0 resources updated in 10 seconds
[2021-03-23T18:27:01-07:00] WARN: Failed to read the private key /etc/chef/client.pem: #<Errno::ENOENT: No such file or directory @ rb_sysopen - /etc/chef/client.pem>
[2021-03-23T18:27:01-07:00] WARN: Error while reporting run start to Data Collector. URL: https://localhost:443/data-collector Exception: No HTTP Code -- I cannot read /etc/chef/client.pem, which you told me to use to sign requests!
[2021-03-23T18:27:01-07:00] WARN: Failed to read the private key /etc/chef/client.pem: #<Errno::ENOENT: No such file or directory @ rb_sysopen - /etc/chef/client.pem>
[2021-03-23T18:27:01-07:00] WARN: Error while reporting run start to Data Collector. URL: https://localhost:443/data-collector Exception: No HTTP Code -- I cannot read /etc/chef/client.pem, which you told me to use to sign requests!
/opt/chef-workstation/embedded/lib/ruby/gems/2.7.0/gems/chef-16.10.17/lib/chef/compliance/runner.rb:23: warning: Chef::Compliance::Runner#logger at /opt/chef-workstation/embedded/lib/ruby/2.7.0/forwardable.rb:154 forwarding to private method NilClass#logger
[2021-03-23T18:27:01-07:00] FATAL: NoMethodError: undefined method `logger' for nil:NilClass
```

**Comment by sebfie** [comment]
+1, also get this error.

**Comment by marcparadise** [comment]
Fixed in #11290
